lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -85,5 +85,13 @@ def sanitize_shop_param(params)
       return unless params[:shop].present?
       ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
     end
+
+    def sanitized_params
+      query_params = request.query_parameters
+      return query_params unless params[:shop].is_a?(String)
+
+      query_params[:shop] = sanitize_shop_param(params)
```

Comment:
This will modify the object that `request.query_parameters` returns, which will affect other consumers of `request.query_parameters` . I suggest we clone the value before modifying it.

Reply:
Excellent point